README.md

```diff
@@ -184,20 +234,63 @@ To run tests and code coverage, use the following command:
+### Adding service dependencies and code into your app
+
+Your service will likely have an SDK library to make programming easier.  Add the service library as a dependency in your package.json file, and add lines of code to initialize and use your new service.  See your service documentation for details.
+
+After that is done, build and run your app locally to test the connection to the service.   Note that by using `ibm cloud edit` to add the service, you get a local credential file and you don't need to add credentials manually to your local environment.
+
+### Modifying cloud deployment assets
+
+After testing your connection locally, run
+
+```bash
+ibmcloud dev edit
+```
+
+again. This time select the "Create manual deployment files" option.  Follow the prompts to create deployment files. The terminal output will tell you which files were created or need to be merged.
+
+For Kubernetes Helm deployments, the values.yaml will require a bit more attention.  The value of `secretKeyRef` will be "undefined", so you will need to add the service credentials as a secret in your Kube cluster, then past the secret name in the values.yml.
+
+### Adding your service credentials as a Kubernetes secret
+
+Copy your service credentials from the IBM Console.  You will find them in the Service Details view for your service instance.
+
+Create a secret yaml file using the service credentials.  You should include all your service credentials as single secret item named 'binding'.  Here is an example yaml file with redacted credential values that you might use as a template:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: node-express-cloudant-secret
+type: Opaque
+stringData:
+  binding: '{"apikey":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","host":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx-bluemix.cloudantnosqldb.appdomain.cloud","iam_apikey_description":"Auto-generated for key xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","iam_apikey_name":"kube-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.xxxxxxxxxxxxxxxxxxxxx.default","iam_role_crn":"crn:v1:bluemix:public:iam::::serviceRole:Manager","iam_serviceid_crn":"crn:v1:bluemix:public:iam-identity::a/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx::serviceid:ServiceId-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","password":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","port":443,"url":"https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx-bluemix:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx-bluemix.cloudantnosqldb.appdomain.cloud","username":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx-bluemix"}'
+```
+
+To add the secret to the cluster, paste your yaml directly into your cluster dashboard yaml editor, or save your secret as yaml file and apply it using your cluster CLI.  For example:
+
+```bash
+kubectl apply -f ./secret.yaml
+```
+
+After creating the secret, don't forget to paste the secret name ("node-express-cloudant-secret" in the example above) into your values.yaml file.  
```

Comment:
Should add a similar sentence to cover Knative, which is defined in the service.yaml file.